conan/cli/args.py

```diff
@@ -56,45 +58,76 @@ def add_common_install_arguments(parser):
 
 
 def add_profiles_args(parser):
-    def profile_args(machine, short_suffix="", long_suffix=""):
-        parser.add_argument("-pr{}".format(short_suffix),
-                            "--profile{}".format(long_suffix),
-                            default=None, action="append",
-                            dest='profile_{}'.format(machine),
-                            help='Apply the specified profile to the {} machine'.format(machine))
-
-    def settings_args(machine, short_suffix="", long_suffix=""):
-        parser.add_argument("-s{}".format(short_suffix),
-                            "--settings{}".format(long_suffix),
-                            action="append",
-                            dest='settings_{}'.format(machine),
-                            help='Settings to build the package, overwriting the defaults'
-                                 ' ({} machine). e.g.: -s{} compiler=gcc'.format(machine,
-                                                                                 short_suffix))
-
-    def options_args(machine, short_suffix="", long_suffix=""):
-        parser.add_argument("-o{}".format(short_suffix),
-                            "--options{}".format(long_suffix),
-                            action="append",
-                            dest="options_{}".format(machine),
-                            help='Define options values ({} machine), e.g.:'
-                                 ' -o{} Pkg:with_qt=true'.format(machine, short_suffix))
-
-    def conf_args(machine, short_suffix="", long_suffix=""):
-        parser.add_argument("-c{}".format(short_suffix),
-                            "--conf{}".format(long_suffix),
+    class ContextAllAction(argparse.Action):
+        def __init__(self,
```

Comment:
Probably args/kwargs can be used to simplify code, both here and in ``super()`` code

Reply:
I wish the append Action was public in argparse, so this could be 
```py
    class ContextAllAction(argparse._AppendAction):
        def __call__(self, action_parser, namespace, values, option_string=None):
            for context in contexts:
                items = getattr(namespace, self.dest + "_" + context, None)
                items = items[:] if items else []
                items.append(values)
                setattr(namespace, self.dest + "_" + context, items)
```
but alas